Sources/Kommunicate/Classes/Kommunicate.swift

```diff
@@ -334,6 +334,23 @@ open class Kommunicate: NSObject, Localizable {
                 completionHandler(false)
                 return
             }
+            // Fetch Chat Context & check for custom bot name in it.If its present, update KMCellConfiguration.customBotName
+            do {
+                if let messageMetadata = channel.metadata as? [String: Any],
+                      let jsonData = messageMetadata[ChannelMetadataKeys.chatContext] as? String,!jsonData.isEmpty,
+                      let data = jsonData.data(using: .utf8),
+                      let chatContextData = try JSONSerialization.jsonObject(with: data, options: JSONSerialization.ReadingOptions.allowFragments) as? [String: Any],
+                   let customBot = chatContextData["bot_customization"] as? [String: String],
+                   let customBotName = customBot["name"],
+                   let customBotId = customBot["id"],
+                   !customBotName.isEmpty, !customBotId.isEmpty {
+                    KMCellConfiguration.customBotName = customBotName
+                    KMCellConfiguration.customizedBotId = customBotId
```

Comment:
Should we also pass this when creating a conversation after pre-chat?

Reply:
yes..After pre chat conversation will be created & opened through this method